### DISCUSSION TITLE
Adding a market sell off function

### DISCUSSION THREAD

**Discussion by Nllii** [steveman1123/multiStonk#2] [Ideas]
Hello, what do you think about adding a trivial function that gets called based on a machine learning or a sentiment analysis function to determine when to  call the
[sellAll function:]

(https://github.com/steveman1123/multiStonk/blob/f5ae03eaee0f25b0f64c07a6920b96cddfbd272e/alpacafxns.py#L75)

Everyday 2 hours before open the script runs and reads articles, and then determines if a market sell off is happening.

Adding a risk level can determine when to resume trading again, at a -3% dip that day, or -5 % of overall portfolio value that day based on the paper trading side of alpaca. 

Instead of a break in the loop after `def sellAll() is called`  the script halts the program? until 1 hour before closing to resume `market_sellAll() ` if it determines the risk tolerance, it then resumes the buying `check2buy` [check2buy function again](https://github.com/steveman1123/multiStonk/blob/f5ae03eaee0f25b0f64c07a6920b96cddfbd272e/multiStonk.py#L193)

**Comment by steveman1123** [comment]
That may be a good idea for an individual algorithm (I'm not sure it would be good for putting in the sellAll function since that supersedes the individual algorithms and at the portfolio level, and it's possible another algorithm could be using the sell-off to its advantage). So when the getList function is called before market opens (during updateLists()), it could perform a check looking for the selloff, and adjust the take-profits and stop-losses accordingly?  
  
Currently the sellAll should only be called if the overall portfolio value drops to a certain % of the max portfolio value reached in the last month (eg if the max was $100k, SL set to 80%, current portfolio value drops to $80k, then sell everything and rethink your life decisions. This way it's possible to avoid losing your shirt completely)

**Reply by Nllii** [reply]
>  So when the getList function is called before market opens (during updateLists()), it could perform a check looking for the selloff, and adjust the take-profits and stop-losses accordingly?

Thanks for clarifying the idea further.   

> Currently the sellAll should only be called if the overall portfolio value drops to a certain % of the max portfolio value reached in the last month (eg if the max was $100k, SL set to 80%, current portfolio value drops to $80k, then sell everything and rethink your life decisions. This way it's possible to avoid losing your shirt completely)

Am glad you added a sellAll function. At least I didn't loose much ,the program stopped   after 12/10/2021, perfect.

**Reply by steveman1123** [reply]
It's saved my butt a couple times too. I do like the idea of doing some ML, however my ML skills are pretty lacking at the moment, but eventually I think it'd be good to incorporate them
